.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,145 @@
+name: Build and upload to PyPI
+
+on:
+  push:
+    tags:
+      - "*"
```

Comment:
You don't want to filter on specific version tags? Come to think that's fine given that we use these for testpypi.

Reply:
I do it in the core repo but I don't particularly feel like enforcing my tag habits on other, happy to change this to whichever pattern the pytng maintainers enjoy though.

Reply:
That's fair.

Reflecting a bit more, I'd also advocate for starting out less restrictive and tightening up if problems arise.